server/src/main/scala/net/wiringbits/actions/SendEmailVerificationTokenAction.scala

```diff
@@ -0,0 +1,71 @@
+package net.wiringbits.actions
+
+import net.wiringbits.apis.models.EmailRequest
+import net.wiringbits.apis.{EmailApi,ReCaptchaApi}
+import net.wiringbits.config.{UserTokensConfig, WebAppConfig}
+import net.wiringbits.repositories.UsersRepository
+import net.wiringbits.util.{EmailMessage, TokenGenerator, TokensHelper}
+import net.wiringbits.validations.{ValidateCaptcha, ValidateEmailIsRegistered, ValidateUserIsNotVerified}
+import net.wiringbits.repositories.models.{UserToken, UserTokenType}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import net.wiringbits.api.models.SendEmailVerificationToken
+import net.wiringbits.repositories.UserTokensRepository
+import java.time.temporal.ChronoUnit
+import java.time.{Clock, Instant}
+import java.util.UUID
+
+
+class SendEmailVerificationTokenAction  @Inject() (
+    usersRepository: UsersRepository,
+    tokenGenerator: TokenGenerator,
+    userTokensConfig: UserTokensConfig,
+    userTokensRepository: UserTokensRepository,
+    webAppConfig: WebAppConfig,
+    emailApi: EmailApi,
+    reCaptchaApi: ReCaptchaApi,
+)(implicit
+    ec: ExecutionContext,
+    clock: Clock
+) {
+
+  def apply(request: SendEmailVerificationToken.Request): Future[SendEmailVerificationToken.Response] = {
+    for {
+      _ <- validations(request)
+      userMaybe <- usersRepository.find(request.email)
+      user = userMaybe.getOrElse(throw new RuntimeException(s"User with email ${request.email} wasn't found"))
+      _ = ValidateUserIsNotVerified(user)
+
+      
+      token = tokenGenerator.next()
+      hmacToken = TokensHelper.doHMACSHA1(token.toString.getBytes(), userTokensConfig.hmacSecret)
+
+      createToken = UserToken
+      .Create(
+        id = UUID.randomUUID(),
+        token = hmacToken,
+        tokenType = UserTokenType.EmailVerification,
+        createdAt = Instant.now(clock),
+        userId = user.id,
+        expiresAt = Instant.now(clock).plus(userTokensConfig.resetPasswordExp.toHours, ChronoUnit.HOURS)
```

Comment:
You're not using the expiration time of email verification here, I think it is in the `UserTokensConfig` object

Reply:
I d o not understand your comment @caiseL 


What I see is that `ChronoUnit.HOURS` can be problematic if we ever set the expiration to seconds/minutes just for testing, hence, I'd suggest to use `seconds` instead.